data_source/aws_ses_active_receipt_rule_set: Panic when no SES Active Receipt Rule Set exists

### Terraform Core Version

1.1.6

### AWS Provider Version

4.31.0

### Affected Resource(s)

Error: Plugin did not respond
│ 
│   with data.aws_vpc.vpc,
│   on datasources.tf line 12, in data "aws_vpc" "vpc":
│   12: data "aws_vpc" "vpc" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with data.aws_ses_active_receipt_rule_set.active_receipt_rule_set,
│   on ses.tf line 1, in data "aws_ses_active_receipt_rule_set" "active_receipt_rule_set":
│    1: data "aws_ses_active_receipt_rule_set" "active_receipt_rule_set" {}
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.

### Expected Behavior

Expected Terraform plan to succeed

### Actual Behavior

AWS provider crashed

### Relevant Error/Panic Output Snippet

```shell
Error: Plugin did not respond
│ 
│   with data.aws_vpc.vpc,
│   on datasources.tf line 12, in data "aws_vpc" "vpc":
│   12: data "aws_vpc" "vpc" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with data.aws_ses_active_receipt_rule_set.active_receipt_rule_set,
│   on ses.tf line 1, in data "aws_ses_active_receipt_rule_set" "active_receipt_rule_set":
│    1: data "aws_ses_active_receipt_rule_set" "active_receipt_rule_set" {}
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Warning: Value for undeclared variable
│ 
│ The root module does not declare a variable named
│ "infrastructure_team_email" but a value was found in file
│ "/build/406320440205/account.tfvars". If you meant to use this value, add a
│ "variable" block to the configuration.
│ 
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
╷
│ Warning: Values for undeclared variables
│ 
│ In addition to the other similar warnings shown, 3 other variable(s)
│ defined without being declared.
╵

Stack trace from the terraform-provider-aws_v4.31.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x742add8]

goroutine 356 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ses.dataSourceActiveReceiptRuleSetRead(0x0?, {0x8de53a0?, 0xc0005e0000})
	github.com/hashicorp/terraform-provider-aws/internal/service/ses/active_receipt_rule_set_data_source.go:39 +0xb8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xaeef8c0?, {0xaeef8c0?, 0xc0020d18f0?}, 0xd?, {0x8de53a0?, 0xc0005e0000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.22.0/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0007fd6c0, {0xaeef8c0, 0xc0020d18f0}, 0xc0032fd100, {0x8de53a0, 0xc0005e0000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.22.0/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000121668, {0xaeef8c0?, 0xc0020d17d0?}, 0xc00341e920)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.22.0/helper/schema/grpc_provider.go:1179 +0x38f
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadDataSource({0xc001ec14a0, 0xc001ec1500, {0xc001eab8e0, 0x2, 0x2}, 0xc001ec14d0, 0xc0008b7a20, 0xc0018ae090, 0xc001ec1530}, {0xaeef8c0, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.7.0/tf5muxserver/mux_server_ReadDataSource.go:27 +0x142
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc002d065a0, {0xaeef8c0?, 0xc0020d0fc0?}, 0xc0022c99a0)
	github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658 +0x3ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x9e78960?, 0xc002d065a0}, {0xaeef8c0, 0xc0020d0fc0}, 0xc00341d8f0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.****************************************_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc002d243c0, {0xaef5160, 0xc002ccd1e0}, 0xc003102c60, 0xc001fafd10, 0x10603dd0, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1295 +0xb0b
google.golang.org/grpc.(*Server).handleStream(0xc002d243c0, {0xaef5160, 0xc002ccd1e0}, 0xc003102c60, 0x0)
	google.golang.org/grpc@v1.48.0/server.go:1636 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.48.0/server.go:930 +0x28a

Error: The terraform-provider-aws_v4.31.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

time=2022-09-22T13:52:39Z level=error msg=1 error occurred:
	* exit status 1
```


### Terraform Configuration Files

```terraform
data "aws_ses_active_receipt_rule_set" "active_receipt_rule_set" {}

data "aws_vpc" "vpc" {
  tags = {
    app-id = var.app_id
  }
}
```

var.app_id = 209486 which does exist in the specified region (us-west-2)

There is **no** active_receipt_rule_set, but I would expect the data source to complain, but not crash.

### Steps to Reproduce

Running a `terraform plan` causes the stack trace above.

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

This is occurring in our production AWS account, but does run in our non-production AWS account. I feel like the problem is the missing active_receipt_rule_set and for some reason that's causing a crash instead of a "not found"

### References

_No response_

### Would you like to implement a fix?

_No response_

--- a/.changelog/27073.txt
+++ b/.changelog/27073.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+data-source/aws_ses_active_receipt_rule_set: Prevent crash when no receipt rule set is active
+```

--- a/internal/service/ses/active_receipt_rule_set_data_source.go
+++ b/internal/service/ses/active_receipt_rule_set_data_source.go
@@ -8,6 +8,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/ses"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 )
 
 func DataSourceActiveReceiptRuleSet() *schema.Resource {
@@ -34,6 +35,9 @@ func dataSourceActiveReceiptRuleSetRead(d *schema.ResourceData, meta interface{}
 
 	if err != nil {
 		return fmt.Errorf("reading SES Active Receipt Rule Set: %s", err)
+	}
+	if data == nil || data.Metadata == nil {
+		return tfresource.NewEmptyResultError(nil)
 	}
 
 	name := aws.StringValue(data.Metadata.Name)

--- a/internal/service/ses/active_receipt_rule_set_data_source_test.go
+++ b/internal/service/ses/active_receipt_rule_set_data_source_test.go
@@ -2,7 +2,6 @@ package ses_test
 
 import (
 	"fmt"
-	"os"
 	"regexp"
 	"testing"
 
@@ -39,12 +38,6 @@ func testAccActiveReceiptRuleSetDataSource_basic(t *testing.T) {
 }
 
 func testAccActiveReceiptRuleSetDataSource_noActiveRuleSet(t *testing.T) {
-	// Gate this test so active SES receipt rule sets are not de-activated unless
-	// explicitly intended
-	if os.Getenv("UNSET_ACTIVE_SES_RECEIPT_RULE_SET") == "" {
-		t.Skip("UNSET_ACTIVE_SES_RECEIPT_RULE_SET environment variable not set")
-	}
-
 	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			acctest.PreCheck(t)

--- a/internal/service/ses/active_receipt_rule_set_data_source_test.go
+++ b/internal/service/ses/active_receipt_rule_set_data_source_test.go
@@ -107,6 +107,4 @@ func testAccPreCheckUnsetActiveRuleSet(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected PreCheck error: %s", err)
 	}
-
-	return
 }

--- a/internal/service/ses/active_receipt_rule_set_data_source_test.go
+++ b/internal/service/ses/active_receipt_rule_set_data_source_test.go
@@ -2,12 +2,15 @@ package ses_test
 
 import (
 	"fmt"
+	"os"
+	"regexp"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/service/ses"
 	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 )
 
 func testAccActiveReceiptRuleSetDataSource_basic(t *testing.T) {
@@ -35,6 +38,30 @@ func testAccActiveReceiptRuleSetDataSource_basic(t *testing.T) {
 	})
 }
 
+func testAccActiveReceiptRuleSetDataSource_noActiveRuleSet(t *testing.T) {
+	// Gate this test so active SES receipt rule sets are not de-activated unless
+	// explicitly intended
+	if os.Getenv("UNSET_ACTIVE_SES_RECEIPT_RULE_SET") == "" {
+		t.Skip("UNSET_ACTIVE_SES_RECEIPT_RULE_SET environment variable not set")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			testAccPreCheck(t)
+			testAccPreCheckUnsetActiveRuleSet(t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, ses.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccActiveReceiptRuleSetDataSourceConfig_noActiveRuleSet(),
+				ExpectError: regexp.MustCompile("empty result"),
+			},
+		},
+	})
+}
+
 func testAccActiveReceiptRuleSetDataSourceConfig_basic(name string) string {
 	return fmt.Sprintf(`
 resource "aws_ses_receipt_rule_set" "test" {
@@ -49,4 +76,37 @@ data "aws_ses_active_receipt_rule_set" "test" {
   depends_on = [aws_ses_active_receipt_rule_set.test]
 }
 `, name)
+}
+
+func testAccActiveReceiptRuleSetDataSourceConfig_noActiveRuleSet() string {
+	return `
+data "aws_ses_active_receipt_rule_set" "test" {}
+`
+}
+
+func testAccPreCheckUnsetActiveRuleSet(t *testing.T) {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).SESConn
+
+	output, err := conn.DescribeActiveReceiptRuleSet(&ses.DescribeActiveReceiptRuleSetInput{})
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if output == nil || output.Metadata == nil {
+		return
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+
+	_, err = conn.SetActiveReceiptRuleSet(&ses.SetActiveReceiptRuleSetInput{
+		RuleSetName: nil,
+	})
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+
+	return
 }

--- a/internal/service/ses/active_receipt_rule_set_test.go
+++ b/internal/service/ses/active_receipt_rule_set_test.go
@@ -23,7 +23,8 @@ func TestAccSESActiveReceiptRuleSet_serial(t *testing.T) {
 			"disappears": testAccActiveReceiptRuleSet_disappears,
 		},
 		"DataSource": {
-			"basic": testAccActiveReceiptRuleSetDataSource_basic,
+			"basic":           testAccActiveReceiptRuleSetDataSource_basic,
+			"noActiveRuleSet": testAccActiveReceiptRuleSetDataSource_noActiveRuleSet,
 		},
 	}
 

